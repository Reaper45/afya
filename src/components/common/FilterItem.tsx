import React from 'react'
import { ButtonHTMLAttributes } from 'react'

import styled from 'lib/emotion'
import Button from "./Button";

import menu from "static/icons/icon-more-horiz.svg";

export const menuIcon = (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
    <path className="heroicon-ui" d="M4 15a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm8 2a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm8 2a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"/>
  </svg>
)

const FilterItemWrapper = styled("button")<{
  outline?: boolean;
  active?: boolean;
}>`
  padding: 0.5rem 0.5rem 0.5rem 1rem;
  background: ${props => (props.active ? props.theme.colors.primary : `#fff`)};
  color: ${props => (props.active ? props.theme.colors.light : props.theme.colors.accent)};
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  border: solid 1px ${props => (props.active ? props.theme.colors.primary : props.theme.colors.grey)};
  border-radius: 4px;
  font-weight: 700;
  font-size: 0.8rem;
  margin-bottom: 1em;
  outline: none;
  padding: .65rem .4rem .65rem 1rem;
  cursor: pointer;
  svg {
    transform: rotate(90deg);
    display: block;
    fill: ${props => props.theme.colors.grey};
  }
  > div {
    flex-grow: 1;
    padding: 0 1rem;
    text-align: left;
  }
  > [data-name="count"] {
    color: ${props => (props.active ? props.theme.colors.light : props.theme.colors.primary)};
  }
`;

export interface IFilterItemProps {
  active?: boolean;
  count?: number;
  title: string;
  description?: string;
  outline?: boolean;
}

const FilterItem: React.FC<ButtonHTMLAttributes<HTMLButtonElement> & IFilterItemProps> = ({ count, title, description, ...props }) => {
  return (
    <FilterItemWrapper {...props}>
      {count && <span data-name="count">{count}</span>}
      <div>
        <span data-name="title">{title}</span>
        {description && <span data-name="description">{description}</span>}
      </div>
      {menuIcon}
    </FilterItemWrapper>
  );
};

export default FilterItem;
