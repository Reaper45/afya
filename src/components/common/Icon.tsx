import * as React from 'react'

import styled from 'lib/emotion'

const IconWrapper = styled("span")<{ active?: boolean }>`
  color: ${props =>
    props.active ? props.theme.colors.accent : props.theme.colors.primary};
`;

const Icon: React.FC<{
  active?: boolean; icon: JSX.Element;
}> = ({ active, icon}) => {
  return <IconWrapper active={active}>{icon}</IconWrapper>;
};

export default Icon;

export const calenderIcon = (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
    <path className="heroicon-ui" d="M17 4h2a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V6c0-1.1.9-2 2-2h2V3a1 1 0 1 1 2 0v1h6V3a1 1 0 0 1 2 0v1zm-2 2H9v1a1 1 0 1 1-2 0V6H5v4h14V6h-2v1a1 1 0 0 1-2 0V6zm4 6H5v8h14v-8z"/>
  </svg>
)

export const chatIcon = (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
    <path className="heroicon-ui" d="M6 14H4a2 2 0 0 1-2-2V4c0-1.1.9-2 2-2h12a2 2 0 0 1 2 2v2h2a2 2 0 0 1 2 2v13a1 1 0 0 1-1.7.7L16.58 18H8a2 2 0 0 1-2-2v-2zm0-2V8c0-1.1.9-2 2-2h8V4H4v8h2zm14-4H8v8h9a1 1 0 0 1 .7.3l2.3 2.29V8z"/>
  </svg>
)

export const moreHorizontalIcon = (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
    <path className="heroicon-ui" d="M4 15a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm8 2a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm8 2a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"/>
  </svg>
)