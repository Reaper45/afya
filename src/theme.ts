export interface ITheme {
  colors: {
    primary: string;
    dark: string;
    light: string;
    secondary: string;
    accent: string;
  };
}

export const theme: ITheme = {
  colors: {
    primary: "#5A67D8",
    dark: "#3C366B",
    light: "#A3BFFA",
    secondary: "#222222",
    accent: "A0AEC0"
  }
};
