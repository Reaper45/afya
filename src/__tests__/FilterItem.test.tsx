import React from "react";
import { shallow, mount } from "enzyme";

import { ThemeProvider } from "emotion-theming";

import { theme } from "theme";
import FilterItem, { IFilterItemProps } from "components/common/FilterItem";
import Icon, { moreHorizontalIcon } from "components/common/Icon";

import icon from "static/icons/menu.svg";

const filter: IFilterItemProps = {
  count: 4,
  title: "Kiambu",
  description: "Test Description",
};

it("renders without crashing", () => {
  const component = shallow(<FilterItem {...filter} />);
  expect(component).toMatchSnapshot();
});

it("renders props correctly", () => {
  const component = shallow(<FilterItem {...filter} />);
  expect(component.find('[data-name="count"]').text()).toEqual("4");
  expect(component.find('[data-name="title"]').text()).toEqual("Kiambu");
  expect(component.find('[data-name="description"]').text()).toEqual(
    "Test Description"
  );
});

it("contains more menu", () => {
  const component = mount(
    <ThemeProvider theme={theme}>
      <FilterItem {...filter} />
    </ThemeProvider>
  );
  expect(component.containsMatchingElement(moreHorizontalIcon)).toBe(true);
});
