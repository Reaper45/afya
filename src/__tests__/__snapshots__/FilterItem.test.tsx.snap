// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Styled(button)>
  <span
    data-name="count"
  >
    4
  </span>
  <div>
    <span
      data-name="title"
    >
      Kiambu
    </span>
    <span
      data-name="description"
    >
      Test Description
    </span>
  </div>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className="heroicon-ui"
      d="M4 15a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm8 2a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm8 2a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"
    />
  </svg>
</Styled(button)>
`;
